src/openforms/registrations/contrib/objects_api/migrations/0011_create_objecttypesypes_service_from_url.py

```diff
@@ -0,0 +1,95 @@
+# Generated by Django 3.2.23 on 2024-01-31 14:20
+import logging
+import re
+from furl import furl
+
+from django.db import migrations
+from django.db.migrations.state import StateApps
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+
+from openforms.registrations.contrib.objects_api.plugin import (
+    PLUGIN_IDENTIFIER as OBJECTS_API_PLUGIN_IDENTIFIER,
+)
+
+from zgw_consumers.constants import APITypes
+
+logger = logging.getLogger(__name__)
+
+
+def create_objecttypes_api_service(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    """Create a "placeholder" Objecttypes API service from an already existing Objecttypes URL.
+
+    This migration exists to help users migrate to a new version of Open Forms. Note that
+    they will still need to configure authentication on the created Objecttypes Service.
+    """
+    ObjectsAPIConfig = apps.get_model("registrations_objects_api", "ObjectsAPIConfig")
+
+    # First, try getting the default objecttypes URL
+    try:
+        objects_api_config = ObjectsAPIConfig.objects.get()
+    except ObjectsAPIConfig.DoesNotExist:
+        return
+    if objects_api_config.objecttypes_service:
+        # Extra safety, even though it shouldn't happen
+        return
+
+    objecttypes_url = objects_api_config.objecttype
+
+    # If no default objecttypes URL was configured, get it from form registration backends:
+    if not objecttypes_url:
+        form_registration_urls: set[str] = set()
+
+        FormRegistrationBackend = apps.get_model("forms", "FormRegistrationBackend")
+        for form_registration_backend in FormRegistrationBackend.objects.filter(
+            backend=OBJECTS_API_PLUGIN_IDENTIFIER
+        ).iterator():
+            if url := form_registration_backend.options.get("objecttype"):
+                form_registration_urls.add(url)
+
+        if len(form_registration_urls) >= 2:
+            logger.warning(
+                "Found %d different Objecttypes URLs in form registration backend options",
+                len(form_registration_urls),
+            )
```

Comment:
it's likely this will happen, and it's not a problem as long as they have a common base (e.g. `https://example.com/objecttypes/api/v2/...`)

I'd only emit a warning if there are different bases found, and you can be very pragmatic here and scan for the first occurence of `/api/`